cloud/store/recover.go

```diff
@@ -0,0 +1,183 @@
+
+		st.log.Info(fmt.Sprintf("found previous config at %s, recovery will be initiated", raftDBPath))
+
+		log := new(raft.Log)
+		var peers []Peer
+		peersMap := make(map[string]Peer, l)
+		for i := f; i <= l; i++ {
+			err = bdb.GetLog(i, log)
+			if err != nil {
+				return err
+			}
+
+			// Unmarshal the MessagePack data into a map
+			var decodedData map[string]interface{}
+			err := msgpack.Unmarshal(log.Data, &decodedData)
+			if err != nil {
+				st.log.Error("unable to unmarshal msgpack from raft.db file err=" + err.Error())
+				continue
+			}
+
+			// Extract the "Servers" array from the map
+			servers, ok := decodedData["Servers"].([]interface{})
+			if !ok {
+				st.log.Error("failed to extract 'Servers' array")
+				continue
+			}
+
+			// Iterate over the array and decode each Peer
+			for _, server := range servers {
+				serverMap, ok := server.(map[string]interface{})
+				if !ok {
+					st.log.Error("Failed to convert server to map")
+					continue
+				}
+
+				var peer Peer
+				if address, ok := serverMap["Address"].(string); ok {
+					peer.Address = address
+				}
+				if id, ok := serverMap["ID"].(string); ok {
+					peer.ID = id
+				}
+				if suffrage, ok := serverMap["Suffrage"].(int64); ok {
+					peer.Suffrage = suffrage
+					peer.NonVoter = suffrage != int64(raft.Voter)
+				}
+
+				existedP := peersMap[peer.Address]
+				if reflect.DeepEqual(existedP, peer) {
+					continue
+				}
+				peersMap[peer.Address] = peer
+				peers = append(peers, peer)
+			}
+		}
+
+		if len(peers) == 0 {
+			return nil
+		}
+
+		configData, err := json.MarshalIndent(peers, "", "  ")
+		if err != nil {
+			return err
+		}
+
+		file, err := os.OpenFile(peerFilePath, os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0o644)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		if _, err := file.Write(configData); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// configViaPeers checks if the peers.json file exists with the provided path
+// if the files exists it creates Raft.Configuration from it
+func (st *Store) configViaPeers(peerFilePath string) (raft.Configuration, error) {
+	var configuration raft.Configuration
+
+	if _, err := os.Stat(peerFilePath); err == nil {
+		st.log.Info(fmt.Sprintf("found %s file, collect peers from previous config...", peerFilePath))
+
+		configuration, err = raft.ReadConfigJSON(peerFilePath)
+		if err != nil {
+			return configuration, fmt.Errorf("raft recovery failed to parse peers.json: %w", err)
+		}
+		if err := os.RemoveAll(peerFilePath); err != nil {
+			return configuration, fmt.Errorf(fmt.Sprintf("recovery failed to delete %s, please delete manually", peerFilePath), err)
+		}
+
+		st.log.Info(fmt.Sprintf("deleted %s file and will proceed with the recovery", peerFilePath))
+	}
+
+	st.log.Info("previous raft", "peers", fmt.Sprintf("%v", configuration.Servers))
+	return configuration, nil
+}
+
+// TODO: at the moment the validation of the provided address happens on the ip level but not ports,
```

Comment:
If we leave a TODO In code we should link it to a JIRA item. This to ensure we don't "lose" todos in the code and instead have cards we can track.